### PR TITLE
Feature/login form input

### PR DESCRIPTION
**Pull request by KashyretsIvanna**
fix: gitignore
feature: button for login fix: 

- env for back urk

feature: conditions fix:
-only css

зображення
зображення
feature:conditions
check: accept button, scroll in conditions container, styles
зображення

feature: welcome page
check: simple welcome page
зображення

### CODE REVIEW

**Comment by lyudad** [comment]
do you think its nice when the button glue to input? and labels? in case I will open such website I would close it immediately

File: apps/freelance/src/redux/services/authApi.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+
+export const authApi = createApi({
+  reducerPath: 'authReducer',
+  baseQuery: fetchBaseQuery({ baseUrl: 'http://localhost:3000/auth' }),
```
Comment (lyudad) [inline apps/freelance/src/redux/services/authApi.ts]: no links in the code Never, or envs or constants

Comment (KashyretsIvanna) [reply apps/freelance/src/redux/services/authApi.ts]: updated to constants

File: apps/freelance/src/main.tsx [hunk 2]
```diff
@@ -17,16 +18,19 @@ const root = ReactDOM.createRoot(
...
-        <ThemeProvider theme={baseTheme}>
-          <BrowserRouter>
-            <GlobalStyle />
-            <App />
-          </BrowserRouter>
-        </ThemeProvider>
-      </Provider>
-    </Suspense>
-  </StrictMode>,
+  <GoogleOAuthProvider clientId="71848933982-oc9dn519u5k0dp3pp0a8h20maaokdap8.apps.googleusercontent.com">
```
Comment (lyudad) [inline apps/freelance/src/main.tsx]: I think its not secure save such data in the code

File: apps/freelance/src/redux/services/authApi.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { BACK_HOST, BACK_PORT, BACK_PROTO } from '@freelance/components';
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+
+export const authApi = createApi({
+  reducerPath: 'authReducer',
+  baseQuery: fetchBaseQuery({
+    baseUrl: `${BACK_PROTO}://${BACK_HOST}:${BACK_PORT}/auth`,
```
Comment (lyudad) [inline apps/freelance/src/redux/services/authApi.ts]: its too complicated, when you will push to server, you will not have port (not really like that but you will not see it), it better use one variable

File: libs/components/src/lib/conditions-card/conditions-card.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import { useNavigate } from 'react-router-dom';
+
+import { CardWrapper, NewButton, NewCard } from './styles';
+
+import 'antd/dist/antd.css';
+
+export function ConditionsCard(): ReactElement {
+  const navigate = useNavigate();
+  const [dis, setDisabled] = useState(false);
+  const card = React.useRef<HTMLDivElement>(null);
```
Comment (lyudad) [inline libs/components/src/lib/conditions-card/conditions-card.tsx]: I dont like refs, and I see only one case when we could have it. so my question is why do we need 2 refs in this component

File: libs/components/src/lib/conditions-card/conditions-card.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  const end = React.useRef<HTMLDivElement>(null);
+  const { t } = useTranslation();
+  const isInViewport2 = useIsInViewport();
+  useEffect(() => {
+    setDisabled(!isInViewport2);
+  }, [isInViewport2]);
+
+  function useIsInViewport() {
+    const [isIntersecting, setIsIntersecting] = useState(false);
```
Comment (lyudad) [inline libs/components/src/lib/conditions-card/conditions-card.tsx]: this line should return error, you setup state after some section happened. even if its magically work, all hooks should be on the top of the component

File: libs/components/src/lib/conditions-card/conditions-card.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        ),
+      [],
+    );
+    useEffect(() => {
+      observer.observe(end.current as HTMLElement);
+
+      return () => {
+        observer.disconnect();
+      };
+    }, [observer]);
```
Comment (lyudad) [inline libs/components/src/lib/conditions-card/conditions-card.tsx]: its look weard, I think you are doing something wrong, too complicated for such simple pages

File: libs/components/src/lib/google-login-button/index.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+export function GoogleLoginButton() {
+  const [addUser] = useAddUserGoogleMutation();
+  const navigate = useNavigate();
+
+  return (
+    <ButtonContainer>
+      <GoogleLogin
+        onSuccess={async credentialResponse => {
+          const response = await addUser(credentialResponse);
```
Comment (lyudad) [inline libs/components/src/lib/google-login-button/index.tsx]: try/catch

Comment (KashyretsIvanna) [reply libs/components/src/lib/google-login-button/index.tsx]: used

File: libs/components/src/lib/conditions-card/conditions-card.tsx (hunk 6 above)
Comment (KashyretsIvanna) [reply libs/components/src/lib/conditions-card/conditions-card.tsx]: fixed

File: libs/components/src/lib/conditions-card/conditions-card.tsx (hunk 5 above)
Comment (KashyretsIvanna) [reply libs/components/src/lib/conditions-card/conditions-card.tsx]: fixed

File: libs/components/src/lib/conditions-card/conditions-card.tsx (hunk 4 above)
Comment (KashyretsIvanna) [reply libs/components/src/lib/conditions-card/conditions-card.tsx]: fixed

File: apps/freelance/src/redux/services/authApi.ts (hunk 3 above)
Comment (KashyretsIvanna) [reply apps/freelance/src/redux/services/authApi.ts]: fixed

File: apps/freelance/src/main.tsx (hunk 2 above)
Comment (KashyretsIvanna) [reply apps/freelance/src/main.tsx]: fixed

File: libs/components/src/lib/conditions-card/conditions-card.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+export function ConditionsCard(): ReactElement {
+  const navigate = useNavigate();
+  const [dis, setDisabled] = useState(false);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    setDisabled(true);
+  }, []);
+
+  function handleScroll(element: HTMLElement) {
```
Comment (lyudad) [inline libs/components/src/lib/conditions-card/conditions-card.tsx]: why need manually handle scroll. it look weared, you don't have animation or 3d library components. just use CSS whatever idea you have

File: libs/components/src/lib/constants/constants.ts [hunk 9]
```diff
@@ -18,3 +18,5 @@ export const profileQ1WorkTo = 'work_to';
 export const profileQ1SuffixUSDH = 'USD per hour';
 
 export const prBarProfileQ2Per = 50;
+
+export const BACK_URL = 'http://localhost:3000';
```
Comment (lyudad) [inline libs/components/src/lib/constants/constants.ts]: send me the list what is enviroment variables, what data should be included there, send in private message, without this list no next PR will not be approved, you don't know theory

File: libs/components/src/lib/conditions-card/conditions-card.tsx (hunk 8 above)
Comment (KashyretsIvanna) [reply libs/components/src/lib/conditions-card/conditions-card.tsx]: added just css

File: libs/components/src/lib/constants/constants.ts (hunk 9 above)
Comment (KashyretsIvanna) [reply libs/components/src/lib/constants/constants.ts]: sent

File: apps/freelance/.env.local [hunk 10]
```diff
@@ -0,0 +1,2 @@
+NX_CLIENT_ID="71848933982-oc9dn519u5k0dp3pp0a8h20maaokdap8.apps.googleusercontent.com"
+BACK_URL="http://localhost:3000"
```
Comment (lyudad) [inline apps/freelance/.env.local]: Снимок экрана 2022-11-15 в 19 58 37

Comment (KashyretsIvanna) [reply apps/freelance/.env.local]: fixed) my fault

**Review by lyudad — APPROVED** [review]

**Pull request merged**